tests/integration/integration_steps/time_series_generation.py

```diff
@@ -13,7 +13,7 @@ def generate_datapoints(num_points: int, days_ago_for_time_range_start = 2) -> l
     if (num_points <= 0):
         raise ValueError("Number of data points must be greater than 0")
     datapoints = []
-    current_time = datetime.datetime.utcnow() - datetime.timedelta(days=days_ago_for_time_range_start)
+    current_time = datetime.datetime.now(datetime.UTC) - datetime.timedelta(days=days_ago_for_time_range_start)
```

Comment:
This results in an "AttributeError: module 'datetime' has no attribute 'UTC'"

What was wrong with datetime.utcnow?

Reply:
Circling back on our resolution: `datetime.utcnow` will be deprecated in a later version of Python so using that method in Python 3.12 results in a warning, but the suggested method using `datetime.UTC` didn't work in Python 3.10.  To resolve this, now we'll use `datetime.timezone.utc` to properly convert the timestamp, which doesn't raise a warning in 3.12 and works in 3.10.